packages/compartment-mapper/src/import-lite.js

```diff
@@ -85,16 +155,56 @@ export const loadFromMap = async (readPowers, compartmentMap, options = {}) => {
         exitModuleImportHook: compartmentExitModuleImportHook,
       },
     );
-    const { compartment, pendingJobsPromise } = link(compartmentMap, {
-      makeImportHook,
-      parserForLanguage,
-      languageForExtension,
-      globals,
-      transforms,
-      moduleTransforms,
-      __shimTransforms__,
-      Compartment,
-    });
+
+    /** @type {ImportNowHookMaker | undefined} */
+    let makeImportNowHook;
+
+    /** @type {Compartment} */
+    let compartment;
+    /** @type {Promise<void>} */
+    let pendingJobsPromise;
+
+    if (behavior.type === 'SYNC') {
+      const { importNowHook: exitModuleImportNowHook, syncModuleTransforms } =
+        behavior.options;
+      makeImportNowHook = makeImportNowHookMaker(
+        /** @type {SyncReadPowers} */ (readPowers),
+        entryCompartmentName,
+        {
+          compartmentDescriptors: compartmentMap.compartments,
+          searchSuffixes,
+          exitModuleImportNowHook,
+        },
+      );
+      ({ compartment, pendingJobsPromise } = link(compartmentMap, {
+        makeImportHook,
+        makeImportNowHook,
+        parserForLanguage,
+        languageForExtension,
+        globals,
+        transforms,
+        syncModuleTransforms,
+        __shimTransforms__,
+        Compartment,
+      }));
+    } else {
+      // sync module transforms are allowed, because they are "compatible"
+      // with async module transforms (not vice-versa)
+      const moduleTransforms = /** @type {ModuleTransforms} */ ({
+        ...behavior.options.syncModuleTransforms,
+        ...behavior.options.moduleTransforms,
+      });
```

Comment:
I think this needs to be removed.  right @naugtur?

Reply:
yes, this was supposed to happen in map-parser.js only, not sure how it got here

Reply:
After pulling it out, a test failed, and I remember why it's here: if someone provides `syncTransforms` _but otherwise does not provide adequate configuration for dynamic requires_, we should merge it into `moduleTransforms`, otherwise `syncTransforms` will be completely ignored.  Ignoring it seems like poor UX; if we wanted to ban `syncTranforms` from non-dynamic-require use cases, then we should throw an exception.

Reply:
@naugtur 🔼 for your acknowledgement

Reply:
Can you create a test covering that? I don't see how they would be ignores as the mapParsers function is doing the same merging as we do here and I don't recall any other place where they're being consumed

Reply:
OK, this is addressed